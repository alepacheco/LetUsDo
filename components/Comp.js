function Home() {
  return <div>Imported</div>;
}

export default Home;