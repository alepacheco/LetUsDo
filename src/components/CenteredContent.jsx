import React from 'react';
import '../styles/components/CenteredContent.scss';

export const CenteredContent = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
