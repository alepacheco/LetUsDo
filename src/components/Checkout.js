import React from 'react';
import StripeCheckout from 'react-stripe-checkout';
import Button from 'react-bootstrap/Button';

export default class Checkout extends React.Component {
    constructor(props) {
        super(props);

        this.onToken = this.onToken.bind(this);
    }

    onToken(token) {
        fetch('/save-stripe-token', {
            method: 'POST',
            body: JSON.stringify(token),
        }).then(response => {
            response.json().then(data => {
                alert(`We are in business, ${data.email}`);
            });
        });
    }

    render() {
        return (
            <StripeCheckout
                name="We Do"
                description={this.props.description || 'Complete task'}
                panelLabel="Pay"
                amount={this.props.amount || 2000}
                currency="GBP"
                token={this.onToken}
                stripeKey="pk_test_TYooMQauvdEDq54NiTphI7jx"
            >
                <Button className="get-it-done-button">Get it done</Button>
            </StripeCheckout>
        )
    }
}