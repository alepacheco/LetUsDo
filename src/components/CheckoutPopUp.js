import React from 'react';
import Modal from 'react-bootstrap/Modal';
import Form from 'react-bootstrap/Form';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CheckoutFormWrapper from './stripe/CheckoutFormWrapper';
import { bindActionCreators } from 'redux';

import '../styles/components/CheckoutPopUp.css';
import * as actions from '../actions/checkoutPopUpActions';

const validateEmail = email => {
  // eslint-disable-next-line no-useless-escape
  const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(String(email).toLowerCase());
}

class CheckoutPopUp extends React.Component {
  constructor(props) {
    super(props);

    this.onChangeEmail = this.onChangeEmail.bind(this);
  }

  onChangeEmail(event) {
    const emailValid = validateEmail(event.target.value);

    this.props.actions.setCheckoutEmailValid(emailValid);
    this.props.actions.setCheckoutEmail(event.target.value);
  }

  render() {
    return (
      <Modal
        show={this.props.show}
        onHide={this.props.onHide}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header closeButton>
          <Modal.Title id="contained-modal-title-vcenter">
            Confirm order
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <div className="task-description-title">
            Task description
          </div>
          <div className="task-description-content">
            {this.props.taskText || 'No description specified'}
          </div>
          <Form>
            <Form.Group controlId="formBasicEmail" className="email-form">
              <Form.Control
                type="email"
                placeholder="Enter email"
                onChange={this.onChangeEmail}
                value={this.props.email} />
            </Form.Group>
          </Form>

          <CheckoutFormWrapper />
        </Modal.Body>

      </Modal>
    );
  }
}

const mapStateToProps = (state) => ({
  taskText: state.taskModal.taskText,
  email: state.checkoutPopUp.email
});

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CheckoutPopUp);

CheckoutPopUp.propTypes = {
  onHide: PropTypes.func,
  taskText: PropTypes.string,
  email: PropTypes.string,
  show: PropTypes.bool,
  actions: PropTypes.shape({
    setCheckoutEmailValid: PropTypes.func,
    setCheckoutEmail: PropTypes.func
  }),
};