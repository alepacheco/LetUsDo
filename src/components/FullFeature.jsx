import React from 'react';
import '../styles/components/FullFeature.scss';
import Icon from '../static/images/undraw_process.svg';

export class FullFeature extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="full-feature">
        <div className="wrapper">
          lorem ipsum dolor sit amet, consectetur adip
          <div className="center-image">
            <Icon width="250" height="250" />
          </div>
        </div>
      </div>
    );
  }
}

export default FullFeature;
