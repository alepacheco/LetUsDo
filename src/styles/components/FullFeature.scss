.full-feature {
  
  .wrapper {
    display: flex;
  }

  .side-image {
    margin: 24px;
    display: inline-block;
  }
  .feature-title {
    font-size: 24px;
    font-weight: bold;
    margin: auto 28px 0 28px;
  }
  .feature-text {
    text-align: justify;
    margin: auto 28px 36px 28px;
    display: inline-block;
  }
}
