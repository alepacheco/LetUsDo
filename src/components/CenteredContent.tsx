import React from 'react';
import 'src/styles/components/CenteredContent.scss';

type CenteredContentProps = {
  reverse?: boolean;
};

export const CenteredContent: React.FC<CenteredContentProps> = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
