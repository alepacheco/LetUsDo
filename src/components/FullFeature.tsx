import React from 'react';
import propTypes from 'prop-types';
import '../styles/components/FullFeature.scss';
import { CenteredContent } from './CenteredContent';

type FullFeatureProps = {
  image?: string;
  imageSide?: string;
  title: string;
  hideImage?: boolean;
  children?: any;
};
export const FullFeature: React.FC<FullFeatureProps> = ({
  image,
  imageSide,
  title,
  children,
  hideImage
}) => (
  <div className="full-feature">
    <CenteredContent reverse={imageSide === 'right'}>
      {image && (
        <div className={`side-image ${hideImage ? 'hide-image' : ''}`}>
          <img src={image} width="250" height="250" />
        </div>
      )}
      <div className={`text-bottom ${hideImage ? 'extra-top-margin' : ''}`}>
        <div>
          <div className="feature-title">{title}</div>
          <div className="feature-text">{children}</div>
        </div>
      </div>
    </CenteredContent>
  </div>
);

FullFeature.propTypes = {
  image: propTypes.string,
  imageSide: propTypes.string,
  title: propTypes.string.isRequired,
  hideImage: propTypes.bool,
  children: propTypes.any
}

export default FullFeature;