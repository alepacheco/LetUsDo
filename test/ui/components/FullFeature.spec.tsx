/* eslint-disable no-undef, import/no-unresolved */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { FullFeature } from 'components/FullFeature';

describe('<FullFeature />', () => {
  it('renders', () => {
    const wrapper = shallow(
      <FullFeature Image={() => <div> img </div>} title="This is the title">
        This is the content inside
      </FullFeature>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
