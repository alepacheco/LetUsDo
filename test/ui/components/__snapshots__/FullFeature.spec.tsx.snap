// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullFeature /> renders 1`] = `
<div
  className="full-feature"
>
  <Component
    reverse={false}
  >
    <div
      className="side-image "
    >
      <img
        height="250"
        src={[Function]}
        width="250"
      />
    </div>
    <div
      className="text-bottom "
    >
      <div>
        <div
          className="feature-title"
        >
          This is the title
        </div>
        <div
          className="feature-text"
        >
          This is the content inside
        </div>
      </div>
    </div>
  </Component>
</div>
`;
